src/curvefit/models/core_model.py

```diff
@@ -0,0 +1,34 @@
+from curvefit.core.objective_fun import objective_fun
+
+class Model:
+
+    def __init__(self, param_set, curve_fun, loss_fun):
+        self.curve_fun = curve_fun
+        self.loss_fun = loss_fun
+
```

Comment:
```
def create_fun(df, data_specs, params):
    data_specs = data[1]
    t = df[data_specs.col_t].to_numpy()
    obs = df[data_specs.col_obs].to_numpy()
    obs_se = df[data_specs.col_obs_se].to_numpy()
    covs_mat = []
    for covs in self.param_set.covariate:
        covs_mat.append(df[covs].to_numpy())
    return objective_fun(...)
```

Reply:
@zhengp0 @jizezhang will the gradient function be different with AD? If so, how will the two different gradient functions interact with all this stuff?

Reply:
maybe the AD thing could go into `solver`? for example we pass an `ad` option to `fit` function, and if none we use the `gradient` from model instance, otherwise do AD on objective. something like that? @zhengp0 ?

Reply:
I think that sounds good to me, what do you think @zhengp0? Definitely think AD should be a `Solver`.